AlpsSDK/ProximityHandler.swift

```diff
@@ -0,0 +1,233 @@
+//
+
+import Foundation
+import CoreLocation
+import Alps
+
+class ProximityHandler {
+    
+    var contextManager: ContextManager!
+    static var immediateTrigger: [String: ProximityEvent] = [:]
+    static var nearTrigger: [String: ProximityEvent] = [:]
+    static var farTrigger: [String: ProximityEvent] = [:]
+    static var unknownTrigger: [String: ProximityEvent] = [:]
+    var immediateBeacons: [String] = []
+    var nearBeacons: [String] = []
+    var farBeacons: [String] = []
+    var unknownBeacons: [String] = []
+    var refreshTimer: Int = 60 * 1000 // timer is in milliseconds
+    
+    init(contextManager: ContextManager) {
+        self.contextManager = contextManager
+    }
+    
+    // Parsing beacons proximity relatively to the device that detect them
+    func parseBeaconsByProximity(_ beacons: [CLBeacon]) {
+        var ourBeacon: IBeaconTriple?
+        for beacon in beacons {
+            let b = syncBeacon(beacon: beacon)
+            if !b.isEmpty {
+                ourBeacon = b[0]
+            }
+            guard let deviceId = ourBeacon?.deviceId else {
+                return
+            }
+            removeDuplicate(clBeacon: beacon, deviceId: deviceId)
+        }
+    }
+    
+    // Launch a proximity event
+    func triggerBeaconsProximityEvent() {
+        for clProximity in CLProximity.allValues {
+            var (beacons, trigger, distance) = setUpTriggerBeaconsProximityEvent(forCLProximity: clProximity)
+            for id in beacons {
+                // Check if a proximity event already exist
+                if trigger[id] == nil {
+                    // Send the proximity event
+                    let proximityEvent = ProximityEvent.init(deviceId: id, distance: distance)
+                    let deviceId = "NEED CHANGE to a guard"
+                        sendProximityEvent(deviceId: deviceId, proximityEvent: proximityEvent) {(_ proximityEvent) in
+                            guard let pe = proximityEvent else {
+                                return
+                            }
+                            self.addProximityEvent(deviceId: id, proximityEvent: pe, clProximity: clProximity)
+                    }
+                } else {
+                    //this id is already triggered and might need to be refresh -> refreshTriggers()
+                }
+            }
+        }
+    }
+    
+    // Re-launch a proximity event depending on refreshTimer (Default : 60 seconds)
+    func refreshTriggers() {
+        for clProximity in CLProximity.allValues {
+            var (beacons, trigger, distance) = setUpTriggerBeaconsProximityEvent(forCLProximity: clProximity)
+            for id in beacons {
+                // Check if the existed proximity event needs a refresh on a based timer
+                let proximityEvent = trigger[id]
+                // Represents  the UNIX current time in milliseconds
+                let now = Int64(Date().timeIntervalSince1970 * 1000)
```

Comment:
Let's add `Date` extension that allows creating nice timestamps like: `Date.nowTimeInterval`